### PR TITLE
Unit tests + minor fixes & refactor

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added unit tests
- minor fixes & refactor in both clients

### CODE REVIEW

File: clients/elrond/client.go [hunk 1]
```diff
@@ -335,32 +336,40 @@ func (c *client) PerformAction(ctx context.Context, actionID uint64, batch *clie
...
+		c.setStatusForAvailabilityCheck(ethElrond.Unavailable, err.Error())
+
 		return err
 	}
-	c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError, "")
 
 	if currentNonce != c.lastNonce {
 		c.retriesAvailabilityCheck = 0
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Available.String())
+		c.setStatusForAvailabilityCheck(ethElrond.Available, "")
```
Comment (dragos-rebegea) [inline clients/elrond/client.go:353]: i would still empty MetricLastElrondClientError independently of the nonce. 
If we had previously one error and next time we query the nonce, it gives it without error, but it is the same as the store one. We would still have the error (that is not valid anymore) stored in the metric

Comment (dragos-rebegea) [reply clients/elrond/client.go:353]: maybe also add a test for this case

File: testsCommon/bridge/ethereumClientWrapperStub.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 
 // EthereumClientWrapperStub -
 type EthereumClientWrapperStub struct {
+	core.StatusHandler
```
Comment (dragos-rebegea) [inline testsCommon/bridge/ethereumClientWrapperStub.go:17]: :+1:

File: clients/elrond/client.go (hunk 1 above)
Comment (iulianpascalau) [reply clients/elrond/client.go:353]: done & done

File: clients/elrond/client.go [hunk 3]
```diff
@@ -335,32 +336,47 @@ func (c *client) PerformAction(ctx context.Context, actionID uint64, batch *clie
...
-	c.retriesAvailabilityCheck++
-	if c.retriesAvailabilityCheck >= c.allowDelta {
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Unavailable.String())
-		c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError,
-			fmt.Sprintf("nonce %d fetched for %d times in a row", currentNonce, c.retriesAvailabilityCheck))
+
+	// if we reached this point we will need to increment the retries counter
+	defer c.incrementRetriesAvailabilityCheck()
+
+	if c.retriesAvailabilityCheck > c.allowDelta {
```
Comment (dragos-rebegea) [inline clients/elrond/client.go:359]:
```suggestion
	if c.retriesAvailabilityCheck >= c.allowDelta {
```

File: clients/elrond/client_test.go [hunk 4]
```diff
@@ -565,3 +591,112 @@ func TestClient_Close(t *testing.T) {
...
+		},
+		GetNetworkStatusCalled: func(ctx context.Context, shardID uint32) (*data.NetworkStatus, error) {
+			currentNonce += incrementor
+			return &data.NetworkStatus{
+				Nonce: currentNonce,
+			}, nil
+		},
+	}
+	c, _ := NewClient(args)
+	t.Run("different current nonce should update - 10 times", func(t *testing.T) {
```
Comment (dragos-rebegea) [inline clients/elrond/client_test.go:616]: use exactly AllowedDelta value

File: clients/elrond/client.go (hunk 3 above)
Comment (iulianpascalau) [reply clients/elrond/client.go:359]: no, not required, the incrementation is done on the defer. Please check unit test :)

File: clients/elrond/client_test.go (hunk 4 above)
Comment (iulianpascalau) [reply clients/elrond/client_test.go:616]: this is a repetitive test. Could have been 100 as well

**Review by sstanculeanu — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**